main/src/ca/uwaterloo/flix/runtime/interpreter/Channel.java

```diff
@@ -48,11 +48,20 @@
    */
   private int bufferSize;
 
+  /**
+   * a flag for whether the channel is unbuffered or not
+   */
+  private boolean unbuffered = false;
+
   public Channel(int bufferSize) {
```

Comment:
Does the weeder enforce that channels have a positive size? or how does that work?

Reply:
It isn't enforced until the Channel constructor is called, which throws an exception if size is less than 0.